plugins/plugin-docker/che-plugin-docker-client/src/main/java/org/eclipse/che/plugin/docker/client/DockerConnector.java

```diff
@@ -1632,25 +1086,178 @@ protected void doStartContainer(String container,
+                                                            .header("Content-Length", entityBytesArray.length)
+                                                            .entity(entityBytesArray)) {
             final DockerResponse response = connection.request();
-            final int status = response.getStatus();
-            if (NO_CONTENT.getStatusCode() != status && NOT_MODIFIED.getStatusCode() != status) {
-                final DockerException dockerException = getDockerException(response);
-                if (OK.getStatusCode() == status) {
-                    // docker API 1.20 returns 200 with warning message about usage of loopback docker backend
-                    LOG.warn(dockerException.getLocalizedMessage());
-                } else {
-                    throw dockerException;
-                }
+            if (response.getStatus() / 100 != 2) {
+                throw getDockerException(response);
+            }
+            return parseResponseStreamAndClose(response.getInputStream(), NetworkCreated.class);
+        }
+    }
+
+    /**
+     * Connects container to docker network
+     *
+     * @throws IOException
+     *         when problems occurs with docker api calls
+     */
+    public void connectContainerToNetwork(String netId, String containerId) throws IOException {
+        connectContainerToNetwork(ConnectContainerToNetworkParams.create(netId, new ConnectContainer().withContainer(containerId)));
+    }
+
+    /**
+     * Connects container to docker network
+     *
+     * @throws IOException
+     *         when problems occurs with docker api calls
+     */
+    public void connectContainerToNetwork(ConnectContainerToNetworkParams params) throws IOException {
+        byte[] entityBytesArray = toJson(params.getConnectContainer()).getBytes(StandardCharsets.UTF_8);
+
+        try (DockerConnection connection = connectionFactory.openConnection(dockerDaemonUri)
+                                                            .method("POST")
+                                                            .path("/networks/" + params.getNetworkId() + "/connect")
+                                                            .header("Content-Type", MediaType.APPLICATION_JSON)
+                                                            .header("Content-Length", entityBytesArray.length)
+                                                            .entity(entityBytesArray)) {
+            final DockerResponse response = connection.request();
+            if (response.getStatus() / 100 != 2) {
+                throw getDockerException(response);
+            }
+        }
+    }
+
+    /**
+     * Disconnects container from docker network
+     *
+     * @throws IOException
+     *         when problems occurs with docker api calls
+     */
+    public void disconnectContainerFromNetwork(@NotNull String netId, @NotNull String containerId) throws IOException {
```

Comment:
Can you pass `null` to a params arg or `ProgressMonitor` args and etc.? Are there `@NotNull`?